uint8 align_mode
bool align_angle
---
string res